File: game_server/ServerProtocol/src/GS2GC/p022_ChatOp/GS2GC_022_001_RetPlayerChatLogin.java
package GS2GC.p022_ChatOp;

import java.nio.ByteBuffer;
public class GS2GC_022_001_RetPlayerChatLogin implements ALBasicProtocolPack._IALProtocolStructure {
/** 聊天系统ID */
private long systemId;
/** 聊天系统标识 */
private String systemTag;
/** 聊天服ip */
private String ip;
/** 聊天服端口 */
private int port;
/** 登录密码 */
private String checkCode;


public GS2GC_022_001_RetPlayerChatLogin() {
	systemId = (long)0;
	systemTag = "";
	ip = "";
	port = 0;
	checkCode = "";
}

public GS2GC_022_001_RetPlayerChatLogin(
	 long _systemId
	, String _systemTag
	, String _ip
	, int _port
	, String _checkCode
) {	systemId = _systemId;
	systemTag = _systemTag;
	ip = _ip;
	port = _port;
	checkCode = _checkCode;
}

public final byte getMainOrder() { return (byte)22; }

public final byte getSubOrder() { return (byte)1; }

/** 聊天系统ID */
public long getSystemId() { return systemId; }
/** 聊天系统ID */
public void setSystemId(long _systemId) { systemId = _systemId; }
/** 聊天系统标识 */
public String getSystemTag() { return systemTag; }
/** 聊天系统标识 */
public void setSystemTag(String _systemTag) { systemTag = _systemTag; }
/** 聊天服ip */
public String getIp() { return ip; }
/** 聊天服ip */
public void setIp(String _ip) { ip = _ip; }
/** 聊天服端口 */
public int getPort() { return port; }
/** 聊天服端口 */
public void setPort(int _port) { port = _port; }
/** 登录密码 */
public String getCheckCode() { return checkCode; }
/** 登录密码 */
public void setCheckCode(String _checkCode) { checkCode = _checkCode; }


public final int GetBufSize() {
	int _size = 12;
	_size += ALBasicProtocolPack.ALProtocolCommon.GetStringBufSize(systemTag);
	_size += ALBasicProtocolPack.ALProtocolCommon.GetStringBufSize(ip);
	_size += ALBasicProtocolPack.ALProtocolCommon.GetStringBufSize(checkCode);

	return _size;
}

public final int GetFullPackBufSize() {
	int _size = 14;
	_size += ALBasicProtocolPack.ALProtocolCommon.GetStringBufSize(systemTag);
	_size += ALBasicProtocolPack.ALProtocolCommon.GetStringBufSize(ip);
	_size += ALBasicProtocolPack.ALProtocolCommon.GetStringBufSize(checkCode);

	return _size;
}



public final void ReadUnzipBuf(ByteBuffer _buf, int _finalPos) {
	 if(_finalPos > 0 && _buf.position() >= _finalPos) return ;
	if(_buf.remaining() > 0) systemId = _buf.getLong();
	 if(_finalPos > 0 && _buf.position() >= _finalPos) return ;
	if(_buf.remaining() > 0) systemTag = ALBasicProtocolPack.ALProtocolCommon.GetStringFromBuf(_buf);
	 if(_finalPos > 0 && _buf.position() >= _finalPos) return ;
	if(_buf.remaining() > 0) ip = ALBasicProtocolPack.ALProtocolCommon.GetStringFromBuf(_buf);
	 if(_finalPos > 0 && _buf.position() >= _finalPos) return ;
	if(_buf.remaining() > 0) port = _buf.getInt();
	 if(_finalPos > 0 && _buf.position() >= _finalPos) return ;
	if(_buf.remaining() > 0) checkCode = ALBasicProtocolPack.ALProtocolCommon.GetStringFromBuf(_buf);
}

public final void PutUnzipBuf(ByteBuffer _buf) {
	_buf.putLong(systemId);
	ALBasicProtocolPack.ALProtocolCommon.PutStringIntoBuf(_buf, systemTag);
	ALBasicProtocolPack.ALProtocolCommon.PutStringIntoBuf(_buf, ip);
	_buf.putInt(port);
	ALBasicProtocolPack.ALProtocolCommon.PutStringIntoBuf(_buf, checkCode);
}

public final ByteBuffer makeFullPackage() {
	int _bufSize = GetBufSize() + 2;
	ByteBuffer _buf = ByteBuffer.allocate(_bufSize);
	_buf.put((byte)22);
	_buf.put((byte)1);
	PutUnzipBuf(_buf);
	_buf.flip();
	return _buf;
}
public final void makeFullPackage(ByteBuffer _recBuf) {
	if(null == _recBuf)
		return ;
	_recBuf.put((byte)22);
	_recBuf.put((byte)1);
	PutUnzipBuf(_recBuf);
}
public final ByteBuffer makePackage() {
	int _bufSize = GetBufSize();
	ByteBuffer _buf = ByteBuffer.allocate(_bufSize);
	PutUnzipBuf(_buf);
	_buf.flip();
	return _buf;
}
public final void readPackage(ByteBuffer _buf) {
	ReadUnzipBuf(_buf, -1);
}
}

File: game_client/Assets/Scripts/ClientProtocol/GS2GC/p022_ChatOp/GS2GC_022_001_RetPlayerChatLogin.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ALBasicProtocolPack;


namespace GS2GC.p022_ChatOp
{

public class GS2GC_022_001_RetPlayerChatLogin : ALBasicProtocolPack._IALProtocolStructure {
/// <summary>
/// 聊天系统ID
/// </summary>
private long systemId;
/// <summary>
/// 聊天系统标识
/// </summary>
private string systemTag;
/// <summary>
/// 聊天服ip
/// </summary>
private string ip;
/// <summary>
/// 聊天服端口
/// </summary>
private int port;
/// <summary>
/// 登录密码
/// </summary>
private string checkCode;


public GS2GC_022_001_RetPlayerChatLogin() {
	systemId = (long)0;
	systemTag = "";
	ip = "";
	port = 0;
	checkCode = "";
}

public GS2GC_022_001_RetPlayerChatLogin(
	long _systemId
	, string _systemTag
	, string _ip
	, int _port
	, string _checkCode
) {	systemId = _systemId;
	systemTag = _systemTag;
	ip = _ip;
	port = _port;
	checkCode = _checkCode;
}

public byte getMainOrder() { return (byte)22; }

public byte getSubOrder() { return (byte)1; }

/// <summary>
/// 聊天系统ID
/// </summary>
public long getSystemId() { return systemId; }
/// <summary>
/// 聊天系统ID
/// </summary>
public void setSystemId(long _systemId) { systemId = _systemId; }
/// <summary>
/// 聊天系统标识
/// </summary>
public string getSystemTag() { return systemTag; }
/// <summary>
/// 聊天系统标识
/// </summary>
public void setSystemTag(string _systemTag) { systemTag = _systemTag; }
/// <summary>
/// 聊天服ip
/// </summary>
public string getIp() { return ip; }
/// <summary>
/// 聊天服ip
/// </summary>
public void setIp(string _ip) { ip = _ip; }
/// <summary>
/// 聊天服端口
/// </summary>
public int getPort() { return port; }
/// <summary>
/// 聊天服端口
/// </summary>
public void setPort(int _port) { port = _port; }
/// <summary>
/// 登录密码
/// </summary>
public string getCheckCode() { return checkCode; }
/// <summary>
/// 登录密码
/// </summary>
public void setCheckCode(string _checkCode) { checkCode = _checkCode; }


public int GetBufSize() {
	int _size = 12;
	_size += ALBasicProtocolPack.ALProtocolCommon.GetStringBufSize(systemTag);
	_size += ALBasicProtocolPack.ALProtocolCommon.GetStringBufSize(ip);
	_size += ALBasicProtocolPack.ALProtocolCommon.GetStringBufSize(checkCode);

	return _size;
}

public int GetFullPackBufSize() {
	int _size = 14;
	_size += ALBasicProtocolPack.ALProtocolCommon.GetStringBufSize(systemTag);
	_size += ALBasicProtocolPack.ALProtocolCommon.GetStringBufSize(ip);
	_size += ALBasicProtocolPack.ALProtocolCommon.GetStringBufSize(checkCode);

	return _size;
}



public void ReadUnzipBuf(ALProtocolBuf _buf, int _finalPos) {
	 if(_finalPos > 0 && _buf.getCurPos() >= _finalPos) return ;
	systemId = _buf.getLong();
	 if(_finalPos > 0 && _buf.getCurPos() >= _finalPos) return ;
	systemTag = _buf.getString();
	 if(_finalPos > 0 && _buf.getCurPos() >= _finalPos) return ;
	ip = _buf.getString();
	 if(_finalPos > 0 && _buf.getCurPos() >= _finalPos) return ;
	port = _buf.getInt();
	 if(_finalPos > 0 && _buf.getCurPos() >= _finalPos) return ;
	checkCode = _buf.getString();
}

public void PutUnzipBuf(ALProtocolBuf _buf) {
	_buf.putLong(systemId);
	_buf.putString(systemTag);
	_buf.putString(ip);
	_buf.putInt(port);
	_buf.putString(checkCode);
}

public byte[] makeFullPackage() {
	int _bufSize = GetBufSize() + 2;
	ALProtocolBuf _buf = ALProtocolBuf.allocate(_bufSize);
	_buf.put((byte)22);
	_buf.put((byte)1);
	PutUnzipBuf(_buf);
	return _buf.getBuf();
}
public void makeFullPackage(ALProtocolBuf _recBuf) {
	if(null == _recBuf)
		return ;
	_recBuf.put((byte)22);
	_recBuf.put((byte)1);
	PutUnzipBuf(_recBuf);
}
public byte[] makePackage() {
	int _bufSize = GetBufSize();
	ALProtocolBuf _buf = ALProtocolBuf.allocate(_bufSize);
	PutUnzipBuf(_buf);
	return _buf.getBuf();
}
public void readPackage(byte[] _buf) {
	ALProtocolBuf _bufObj = new ALProtocolBuf(_buf);
	ReadUnzipBuf(_bufObj, -1);
}
public void readPackage(ALProtocolBuf _buf) {
	ReadUnzipBuf(_buf, -1);
}
public override string ToString() {
	System.Text.StringBuilder builder = new System.Text.StringBuilder();

	builder.Append("{");
	builder.Append("systemId").Append(":").Append(systemId.ToString()).Append(", ");
	builder.Append("systemTag").Append(":").Append(systemTag.ToString()).Append(", ");
	builder.Append("ip").Append(":").Append(ip.ToString()).Append(", ");
	builder.Append("port").Append(":").Append(port.ToString()).Append(", ");
	builder.Append("checkCode").Append(":").Append(checkCode.ToString()).Append(", ");
	builder.Append("}");
	return builder.ToString();
}

}

}

